File: GithubGists/app/src/main/java/ru/avystavkin/githubgists/screen/base/holders/BaseUserHolder.java
package ru.avystavkin.githubgists.screen.base.holders;

import android.text.TextUtils;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import butterknife.BindView;
import ru.avystavkin.githubgists.R;
import ru.avystavkin.githubgists.models.local.User;
import ru.avystavkin.githubgists.utils.Images;

public abstract class BaseUserHolder extends BaseViewHolder {

    @BindView(R.id.gist_user_name)
    protected TextView mUserName;

    @BindView(R.id.gist_img)
    protected ImageView mImageView;

    public BaseUserHolder(View itemView) {
        super(itemView);
    }

    public void bind(@NonNull User user) {
        String userName = user.getName();
        if (!TextUtils.isEmpty(userName))
            mUserName.setText(userName);

        String url = user.getUrl();
        if (!TextUtils.isEmpty(url))
            Images.loadImage(mImageView, url);
    }
}

File: GithubGists/app/src/main/java/ru/avystavkin/githubgists/screen/base/holders/BaseUserHolder.kt
package ru.avystavkin.githubgists.screen.base.holders

import android.view.View
import kotlinx.android.synthetic.main.item_gist.view.*
import ru.avystavkin.githubgists.models.local.User
import ru.avystavkin.githubgists.utils.Images

abstract class BaseUserHolder(itemView: View) : BaseViewHolder(itemView) {

    fun bind(user: User) {
        if (!user.name.isNullOrEmpty()) {
            itemView.tvGistUsername.text = user.name!!
        }

        if (!user.url.isNullOrEmpty()) {
            Images.loadImage(itemView.ivGistImg, user.url!!)
        }
    }
}
